build/tools/SharedCodeGenerator/VersionInfo.cs

```diff
@@ -0,0 +1,67 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Oryx.SharedCodeGenerator
+{
+    public class VersionInfo : IComparable<VersionInfo>
+    {
+        private readonly List<string> nonSemverPlatforms = new List<string> { "golang" };
+
+        public VersionInfo(string displayVersion, string platform)
+        {
+            this.DisplayVersion = displayVersion;
+            this.Platform = platform;
+            if (this.nonSemverPlatforms.Contains(platform))
+            {
+                this.Version = new Version(this.DisplayVersion);
+            }
+            else
+            {
+                this.SemanticVersion = ToSemanticVersion(this.DisplayVersion);
+            }
+        }
+
+        public string DisplayVersion { get; }
+
+        public string Platform { get; }
+
+        public SemVer.Version SemanticVersion { get; }
+
+        public Version Version { get; }
+
+        public int CompareTo(VersionInfo other) =>
+            this.nonSemverPlatforms.Contains(this.Platform)
+            ? this.Version.CompareTo(other.Version)
+            : this.SemanticVersion.CompareTo(other.SemanticVersion);
+
+        private static SemVer.Version ToSemanticVersion(string displayVersion)
+        {
+            var semanticVersionStr = displayVersion;
+
+            // The display version is in preview version format or it's a non-preview version
+            // Both cases can be handled by SemVer library.
+            // Throws ArgumentException if SemVer library found invalid version format.
+            if (displayVersion.Contains('-') || !displayVersion.Any(c => char.IsLetter(c)))
+            {
+                if (Version.TryParse(semanticVersionStr, out var dotnetVersion))
+                {
+                    return new SemVer.Version(dotnetVersion.ToString(3));
```

Comment:
@cormacpayne  So this is a little strange, since the library overrides the built in ToString method. Its behavior is that the argument passed in actually determines the depth of release version to print. See attached picture for details. So what is actually happening here is that we are taking any string that can be parsed by the built in library, but might not be able to be parsed by semver library, and converting it to something that the semver library can parse. 

That being said, you raise a valid point about why this block is even there, and it was leftover from before we handled golang as non-semver. Now, we should not need this since that is being handled appropriately. I will remove.

![image](https://user-images.githubusercontent.com/107068277/181082366-923afdd3-127b-4c64-86d3-d6eecc22a6b7.png)

Reply:
Oops, read `.ToString` as `.Substring` 😄 Thanks for clarifying this -- if it was working before, then I think we're OK to keep it here.

Reply:
I removed it anyways since it just added code that made things more complicated, and was completely unnecessary haha